### PR TITLE
[1989] Funding - UI - confirmation page

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

https://trello.com/c/m9iZr9Jt/1989-m-funding-ui-confirmation-page

### Changes proposed in this pull request

* New component to show training initiative and bursary

### Guidance to review

* Add funding details on trainee, go directly to URL `/trainees/:slug/funding/confirm`

### CODE REVIEW

**Comment by mooktakim** [comment]
[[M] Funding - UI - confirmation page](https://trello.com/c/m9iZr9Jt/1989-m-funding-ui-confirmation-page)

File: app/components/confirmation/funding/view.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        rows
+      end
+
+    private
+
+      def course_subject_one
+        trainee.course_subject_one
+      end
+
+      def show_bursary_funding?
```
Comment (AbigailMcP) [inline app/components/confirmation/funding/view.rb:49]: **This logic needs updating to match the prototype:**

1. If there is no bursary available for the route i.e. `!CalculateBursary.available_for_route?(route)`, then this row is missing.

2. If there is a bursary available for this route i.e. `CalculateBursary.available_for_route?(route)`, then this row is present.

If (2) and the user has selected `applying_for_bursary`, then show the amount as you've done here.

If (2) and the user has not selected `applying_for_bursary`, then show "Not bursary funded" i.e. there was a bursary available but the user has decided that they don't qualify.

Screenshot 2021-06-25 at 14 29 31

**Review by AbigailMcP — COMMENTED** [review]
I haven't gone through all of this yet, but noticed the below:

File: app/components/confirmation/funding/view.rb (hunk 1 above)
Comment (AbigailMcP) [reply app/components/confirmation/funding/view.rb:49]: At the moment, you're showing the bursary row and a bursary amount even if the user has selected not applying for a bursary.

File: app/components/confirmation/funding/view.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        bursary_amount.present?
+      end
+
+      def bursary_amount
+        @bursary_amount ||= CalculateBursary.for_route_and_subject(trainee.training_route.to_sym, course_subject_one)
+      end
+
+      def training_initiative
```
Comment (AbigailMcP) [inline app/components/confirmation/funding/view.rb:57]: This and the bursary row will need fallback content "Not provided" for current non-draft trainees that were never asked the funding questions.

**Comment by AbigailMcP** [comment]
Also, when I head to the confirm page I get this:

Screenshot 2021-06-25 at 14 53 54

Locally I get the error
```
uninitialized constant Trainees::Funding::ConfirmDetailsController Did you mean? Trainees::ConfirmDetailsController Trainees::CourseDetailsController
```

**Comment by FriedSock** [comment]
I left a TODO in the bursary feature spec as it would error because the confirmation page didn't exist https://github.com/DFE-Digital/register-trainee-teachers/blob/d0cb5a966a854a200bd6aa0525cd3abac75c4c8f/spec/features/form_sections/teacher_training_data/edit_bursary_spec.rb#L19 Can we remove this TODO now?

File: app/helpers/confirm_details_helper.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 module ConfirmDetailsHelper
   def trainee_section_update_path(section_key, trainee)
     routes = {
+      "diversity" => "trainee_diversity_confirm_path",
```
Comment (AbigailMcP) [inline app/helpers/confirm_details_helper.rb:6]: Not sure you meant to add this?

File: config/locales/en.yml [hunk 4]
```diff
@@ -519,6 +519,13 @@ en:
         summary_title: Schools
         lead_school: Lead school
         employing_school: Employing school
+    funding:
+      view:
+        title: "Funding"
```
Comment (AbigailMcP) [inline config/locales/en.yml:524]: For consistency, none of these translations need to be in quotes.

**Review by AbigailMcP — APPROVED** [review]
Just a couple of minor comments, but then gtg!

File: app/helpers/confirm_details_helper.rb (hunk 3 above)
Comment (mooktakim) [reply app/helpers/confirm_details_helper.rb:6]: @AbigailMcP Due to the changes in the routes file, this kept throwing error, seemed to be missing from the list.

**Pull request merged**